File: selfstudy/golang/stars_max.go
/*
task 282:
Stars - Max
You need to find the maximum value among star luminosities
*/

package main

import (
	"fmt"
	"reflect"
)

func HighestStars(brightness []int) int {
	if len(brightness) == 0 { return 0 }
	m := brightness[0]
	for _, v := range brightness {
		if v > m { m = v }
	}
	return m
}

func main() {
	tests := []struct {
		input []int
		expected int
	}{
		{[]int{10, 20, 30}, HighestStars([]int{10, 20, 30})},
		{[]int{50, 60, 70}, HighestStars([]int{50, 60, 70})},
		{[]int{1, 2, 3}, HighestStars([]int{1, 2, 3})},
		{[]int{100, 200}, HighestStars([]int{100, 200})},
		{[]int{}, HighestStars([]int{})},
	}
	
	for idx, t := range tests {
		res := HighestStars(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// finally works

File: selfstudy/js/stars_max.js
/*
task 82:
Stars - Max
You need to find the maximum value among star luminosities
*/

function highest_stars(brightness) {
  return brightness.length ? Math.max(...brightness) : 0;
}

// Tests
const tests = [
  { input: [10, 20, 30], expected: highest_stars([10, 20, 30]) },
  { input: [50, 60, 70], expected: highest_stars([50, 60, 70]) },
  { input: [1, 2, 3], expected: highest_stars([1, 2, 3]) },
  { input: [100, 200], expected: highest_stars([100, 200]) },
  { input: [], expected: highest_stars([]) }
];

if (process.argv.includes('mocha')) {
  const assert = require('assert');
  describe('highest_stars', () => {
    tests.forEach((t, idx) => {
      it(`test case ${idx+1}`, () => {
        assert.deepStrictEqual(highest_stars(t.input), t.expected);
      });
    });
  });
} else {
  tests.forEach((t, idx) => {
    console.log(`Test ${idx+1}:`, highest_stars(t.input) === t.expected || JSON.stringify(highest_stars(t.input)) === JSON.stringify(t.expected) ? "PASS" : "FAIL");
  });
}

// works i think
